climakitae/indices.py

```diff
@@ -1,6 +1,73 @@
-"""Functions for deriving fire indices"""
+"""Functions for deriving indices"""
 
 import xarray as xr
+import numpy as np
+
+
+def noaa_heat_index(T, RH):
+    """Compute the NOAA Heat Index
+
+    Parameters
+    ----------
+    T: xr.DataArray
+        Temperature in deg F
+    RH: xr.DataArray
+        Relative Humidity in percentage (0-100)
+
+    Returns
+    --------
+    xr.DataArray
+        Heat index per timestep
+
+    References
+    -----------
+    https://www.wpc.ncep.noaa.gov/html/heatindex_equation.shtml
+
+    """
+    T = T.reindex_like(RH)  # Need to have the same dimension/coordinate orders
+    HI = (
+        -42.379
+        + 2.04901523 * T
+        + 10.14333127 * RH
+        - 0.22475541 * T * RH
+        - 0.00683783 * T * T
+        - 0.05481717 * RH * RH
+        + 0.00122874 * T * T * RH
+        + 0.00085282 * T * RH * RH
+        - 0.00000199 * T * T * RH * RH
+    )
+
+    # Adjust for high temperature, low relative humidity
+    # 80 < T < 112 (deg F)
+    # RH < 13%
+    adj_highT_lowRH = ((13 - RH) / 4) * ((17 - abs(T - 95)) / 17) ** (
+        1 / 2
+    )  # Adjustment
+    HI_highT_lowRH = HI - adj_highT_lowRH  # Subtract adjustment from HI
+
+    # Adjust for low temperature, high relative humidity
+    # 80 < T < 87 (deg F)
+    # RH > 85%
+    adj_lowT_highRH = ((RH - 85) / 10) * ((87 - T) / 5)  # Adjustment
+    HI_lowT_highRH = HI + adj_lowT_highRH  # Add adjustment from HI
+
+    # Use different equation if heat index if the heat index value < 80
+    low_HI = 0.5 * (T + 61.0 + ((T - 68.0) * 1.2) + (RH * 0.094))
+
+    # Adjust heat index depending on different condions for RH, T, and valid range of HI
+    HI = xr.where((RH < 13) & (T > 80), HI_highT_lowRH, HI)
+    HI = xr.where(((RH > 85) & (T < 87) & (T > 80)), HI_lowT_highRH, HI)
+    HI = xr.where(HI < 80, low_HI, HI)
+    HI = xr.where(
```

Comment:
Do we have a reference for this piece for adjustments for temps above 112 deg F being set to nan, or is there a different equation to be used in such cases? Considering LA county might get to 112+ _today_ it definitely is within reason to consider such a scenario now (I can help on this if needed!)

Reply:
Ooo good point. There was a lot of confusing wording in the NOAA page that described how to compute it. I misread it and assumed it was invalid above 112 degF, but I'm realizing that just means I don't need to apply an adjustment factor. Good catch!! Will fix that now.

Here's the reference doc if it's of interest: https://www.wpc.ncep.noaa.gov/html/heatindex_equation.shtml